src/lib/oper_data.c

```diff
@@ -0,0 +1,95 @@
+/* SPDX-License-Identifier: AGPL-3.0-or-later */
+/*
+ * Authors:     Amjad Daraiseh, adaraiseh@okdanetworks.com>
+ *
+ *              This program is free software; you can redistribute it and/or
+ *              modify it under the terms of the GNU Affero General Public
+ *              License Version 3.0 as published by the Free Software Foundation;
+ *              either version 3.0 of the License, or (at your option) any later
+ *              version.
+ *
+ * Copyright (C) 2024 Okda Networks, <contact@okdanetworks.com>
+ */
+
+#include "oper_data.h"
+#include "cmdgen.h"
+
+/**
+ * Struct store yang modules names and their iproute2 show cmd start.
+ */
+struct module_sh_cmdstart {
+    const char *module_name; // Stores Module name
+    const char *showcmd_start; // Store iproute2 show command related to the module
+} ipr2_sh_cmdstart[] = { { "iproute2-ip-link", "ip link show" },
```

Comment:
why not using this to store `oper_pull_path` (if really needed) instead of `ipr2_ip_modules[]` ?

Reply:
discussed this, agreed on the following:
- oper_pull_path remains in iproute2_sysrepo new struct (yang_module)
- when oper_data pull is fully implemented (next PR) we will remove struct module_sh_cmdstart and migrate to use yang extensions.

This PR is intended to introduce the structural changes on iproute2_sysrepo, oper_data_pull code will get updated on next PR with full feature implementation.